examples/plot_compare_calibration.py

```diff
@@ -0,0 +1,119 @@
+"""
+=======================================
+Comparion of Calibration of Classifiers
+=======================================
+
+Well calibrated classifiers are probabilistic classifiers for which the output
+of the predict_proba method can be directly interpreted as a confidence level.
+For instance a well calibrated (binary) classifier should classify the samples
+such that among the samples to which it gave a predict_proba value close to
+0.8, approx. 80% actually belong to the positive class.
+
+LogisticRegression returns well calibrated predictions as it directly
+optimizes log-loss. In contrast, the other methods return biased probilities,
+with different biases per method::
+
+ * GaussianNaiveBayes tends to push probabilties to 0 or 1 (note the counts in
+the histograms). This is mainly because it makes the assumption that features
+are conditionally independent given the class, which is not the case in this
+dataset which contains 2 redundant features.
+
+ * RandomForestClassifier shows the opposite behavior: the histograms show
+peaks at approx. 0.2 and 0.9 probability, while probabilities close to 0 or 1
+are very rare. An explanation for this is given by Niculescu-Mizil and Caruana
+[1]: "Methods such as bagging and random forests that average predictions from
+a base set of models can have difficulty making predictions near 0 and 1
+because variance in the underlying base models will bias predictions that
+should be near zero or one away from these values. Because predictions are
+restricted to the interval [0,1], errors caused by variance tend to be one-
+sided near zero and one. For example, if a model should predict p = 0 for a
+case, the only way bagging can achieve this is if all bagged trees predict
+zero. If we add noise to the trees that bagging is averaging over, this noise
+will cause some trees to predict values larger than 0 for this case, thus
+moving the average prediction of the bagged ensemble away from 0. We observe
+this effect most strongly with random forests because the base-level trees
+trained with random forests have relatively high variance due to feature
+subseting." As a result, the calibration curve shows a characteristic sigmoid
+shape, indicating that the classifier could trust its "intuition" more and
+return probabilties closer to 0 or 1 typically.
+
+* Support Vector Classification (SVC) shows a similar sigmoid curve as
+the  RandomForestClassifier, which is typical for maximum-margin methods
+(compare Niculescu-Mizil and Caruana [1]).
```

Comment:
I think this is confusing: `SVC(probability=True)` is already using the internal Platt-scaling implemented in libsvm which is supposed to attenuate the margin effect. I think we should either add a note here to state that `SVC(probability=True)` are already calibrated but this does not seem to work correctly on this data, or use the uncalibrated decision function of `LinearSVC` to illustrate this point in this example.